pkg/federate/kubefed/resource.go

```diff
@@ -1,17 +1,120 @@
 package kubefed
 
 import (
+	"context"
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	ctlutil "sigs.k8s.io/kubefed/pkg/controller/util"
+	kubefedopt "sigs.k8s.io/kubefed/pkg/kubefedctl/options"
 )
 
 const (
-//	federatedKindPrefix string = "Federated"
+	federatedKindPrefix         = "Federated"
+	errorSettingFederatedFields = "error setting fields necessary for federating: %v"
 )
 
+var systemMetadataFields = []string{"selfLink", "uid", "resourceVersion", "generation", "creationTimestamp", "deletionTimestamp", "deletionGracePeriodSeconds"}
+
+type UnstructuredConversionError struct {
```

Comment:
Do you intend this to be exported?

Reply:
Yes. I want the consumer to have access to it, in case they want to use it.